_GNU_SOURCE redefined

Hello,

While working on [rules_nim](https://github.com/kczulko/rules_nim) I've noticed the following redefinition
https://github.com/nim-lang/Nim/blob/86d6f71f5afd03ce372ee89b348797f43899579a/lib/pure/strutils.nim#L2011

in the build output log:
```
INFO: From Compiling src/rules_nim_nimble_gen_cc_files_only/@m..@s..@s..@s..@sexternal@srules_nim~~nim~nim_linux_x64@slib@spure@sstrutils.nim.c:
bazel-out/k8-fastbuild/bin/src/rules_nim_nimble_gen_cc_files_only/@m..@s..@s..@s..@sexternal@srules_nim~~nim~nim_linux_x64@slib@spure@sstrutils.nim.c:6: warning: "_GNU_SOURCE" redefined
    6 | #define _GNU_SOURCE
      |
In file included from bazel-out/k8-fastbuild/bin/src/rules_nim_nimble_gen_cc_files_only/@m..@s..@s..@s..@sexternal@srules_nim~~nim~nim_linux_x64@slib@spure@sstrutils.nim.c:4:
external/rules_nim~~nim~nim_linux_x64/lib/nimbase.h:78: note: this is the location of the previous definition
   78 | #  define _GNU_SOURCE 1
      |
``` 

Can I add `#ifndef` around it?

Best regards,
Karol

--- a/lib/pure/strutils.nim
+++ b/lib/pure/strutils.nim
@@ -2009,7 +2009,9 @@ func find*(s: string, chars: set[char], start: Natural = 0, last = -1): int {.
 
 when defined(linux):
   proc memmem(haystack: pointer, haystacklen: csize_t,
-              needle: pointer, needlelen: csize_t): pointer {.importc, header: """#define _GNU_SOURCE
+              needle: pointer, needlelen: csize_t): pointer {.importc, header: """#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
 #include <string.h>""".}
 elif defined(bsd) or (defined(macosx) and not defined(ios)):
   proc memmem(haystack: pointer, haystacklen: csize_t,

